src/analysis/processing/qgsalgorithmtransform.cpp

```diff
@@ -107,6 +115,12 @@ QgsFeatureList QgsTransformAlgorithm::processFeature( const QgsFeature &f, QgsPr
   if ( feature.hasGeometry() )
   {
     QgsGeometry g = feature.geometry();
+
+    if ( !mKeepCurvedGeometries )
```

Comment:
Shouldn't we only do this if the crs IS changed? While it may seem silly to use the transform algorithm for no crs change, a processing model may include this step as a way of gauranteeing that any input is always in a particular crs.